### PR TITLE
Update implementation and user stories in DG

### PR DESCRIPTION
**Pull request by siqiang-ng**
Can be reviewed now.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -409,24 +401,25 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | `* * *`  | user                                       | create a tag                                     | find or sort the contacts based on tags later on.                      |
 | `* * *`  | user                                       | delete a tag                                     | remove tags that are accidentally added or no longer required          |
-| `* * *`  | user                                       | list all tags                                    | quickly find all the tags that I have add                              |
-| `* * *`  | user                                       | search for a tag                                 | find all contacts with that tag                                        |
+| `* * *`  | user                                       | find a tag by name                               | locate details of the tag without having to go through the entire list |
+| `* * *`  | user                                       | list all tags                                    | quickly browse through all the tags that I have adde                   |
```
Comment (yeo-kexin) [inline docs/DeveloperGuide.md]: typo: quickly browse through ... have adde`d`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -409,24 +401,25 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* * *`  | student                        	        | add tasks for each module tag                    | remind myself of all the tasks that are under those tags.              |
```
Comment (yeo-kexin) [inline docs/DeveloperGuide.md]: should we state module tag or just tag?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -409,24 +401,25 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | `* *`    | student                        	        | mark the status of a particular task as done     | keep track of the different task progress under that tag.              |
-| `* *`    | student with many project groups	        | add the meeting platform links to each module tag| conveniently contact the team or initiate a team meeting               |
-| `* *`    | student with many project groups	        | delete the meeting platform link for each tag    | remove the link if it has been added wrongly                           |
 | `* *`    | student with many project groups	        | add remarks on the module tag                    | find module-related information from the module tag                    |
 | `* *`    | student with many project groups	        | delete the comments on the module tag            |                                                                        |
 | `*`      | long-time user	                            | archive old module tags 	                       | keep my contacts up to date                                            |
```
Comment (yeo-kexin) [inline docs/DeveloperGuide.md]: should we remove user stories like this?

**Review by yeo-kexin — COMMENTED** [review]
some comments, but otherwise lgtm

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (siqiang-ng) [reply docs/DeveloperGuide.md]: i tink just tag

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (siqiang-ng) [reply docs/DeveloperGuide.md]: I think since one star means unlikely to have it is alright for them to be still there?

Comment (yeo-kexin) [reply docs/DeveloperGuide.md]: ah okie!

**Review by JoeyToh — APPROVED** [review]
lgtm

**Pull request merged**